glib-build-tools/src/lib.rs

```diff
@@ -57,3 +62,67 @@ pub fn compile_resources<P: AsRef<Path>>(source_dirs: &[P], gresource: &str, tar
         println!("cargo:rerun-if-changed={dep}");
     }
 }
+
+// rustdoc-stripper-ignore-next
+/// Call to run `glib-compile-schemas` to generate compiled gschemas from `.gschema.xml` schemas
+/// files from specified directories and store `gschemas.compiled` into `glib-2.0` schema cache
+/// directory.
+///
+/// ```no_run
+/// glib_build_tools::compile_schemas(
+///     &["schemas"],
+/// );
+/// ```
+pub fn compile_schemas(schemas_dir: &[&str]) {
+    // Target Directory
+    //  Unix - $HOME/.local/share/glib-2.0/schemas/
+    //  Windows - C:/ProgramData/glib-2.0/schemas/
+
+    let prefix = if cfg!(windows) {
+        PathBuf::from("C:/ProgramData")
+    } else {
+        glib::user_data_dir()
+    };
```

Comment:
This hardcoding the prefix to use is meh. Either the prefix will have to be passed as a param or this function won't land as is.